### ISSUE TITLE
train so slowly

### ISSUE THREAD

**Issue by xuanzhangyang** [jiny2001/dcscn-super-resolution#10]
I train your code use yang_bsd_4 dataset, it's so slowly. suggest use tfrecords for loading data, and tf.depth_to_space can replace convert_from_multi_channel_image, tf.spaces_to_depth can replace convert_to_multi_channel_image imilarly. 
These can improve train speed.

**Comment by jiny2001** [comment]
Hi, 

You are totally right.

I had read the papers from NTIRE SR challenge and SR-Res Net from twitter. And my own recent experimental code has those technics they are using (Self Ensemble, Pixel Shuffler and so on). I'm definitely using tf.spaces_to_depth() as you mentioned in my recent code. :)

However, there are lots of changes from the code I published on github, and my experimental code contains other technics I'm just trying. So I'm little hesitate to upload that.

Since there is some improvement on PSNR, I'd like to update those source codes. But now the code is little dirty and experimental. Please wait a while until I can have a time to do that.

Jin

**Comment by jiny2001** [comment]
I updated the codes to use tf.spaces_to_depth. And other improvements are there. Please try it.

Later I will implement pre-processing training data feature which makes training much faster.
